pkg/kubecost/allocation.go

```diff
@@ -506,28 +550,49 @@ func (a *Allocation) add(that *Allocation) {
 	a.CPUCoreHours += that.CPUCoreHours
 	a.GPUHours += that.GPUHours
 	a.RAMByteHours += that.RAMByteHours
-	a.PVByteHours += that.PVByteHours
 
 	// Sum all cumulative cost fields
 	a.CPUCost += that.CPUCost
 	a.GPUCost += that.GPUCost
 	a.RAMCost += that.RAMCost
-	a.PVCost += that.PVCost
 	a.NetworkCost += that.NetworkCost
 	a.LoadBalancerCost += that.LoadBalancerCost
 	a.SharedCost += that.SharedCost
 	a.ExternalCost += that.ExternalCost
 
+	// Sum PV Breakdown
+	a.AddPVBreakDown(that.PVBreakdown)
+
 	// Sum all cumulative adjustment fields
 	a.CPUCostAdjustment += that.CPUCostAdjustment
 	a.RAMCostAdjustment += that.RAMCostAdjustment
 	a.GPUCostAdjustment += that.GPUCostAdjustment
+	a.PVCostAdjustment += that.PVCostAdjustment
 
 	// Any data that is in a "raw allocation only" is not valid in any
 	// sort of cumulative Allocation (like one that is added).
 	a.RawAllocationOnly = nil
 }
 
+// AddPVBreakDown adds contents of pvBreakdown to Allocation PVBreakdown
+// Property, creating new PVUsage where necessary
+func (a *Allocation) AddPVBreakDown(pvBreakdown map[string]PVUsage) {
+	if pvBreakdown != nil {
+		if a.PVBreakdown == nil {
+			a.PVBreakdown = map[string]PVUsage{}
+		}
+		for pvName, pv := range pvBreakdown {
+			apv, ok := a.PVBreakdown[pvName]
+			if !ok {
+				apv = PVUsage{}
+			}
+			apv.Cost += pv.Cost
+			apv.ByteHours += pv.ByteHours
+			a.PVBreakdown[pvName] = apv
+		}
+	}
+}
```

Comment:
I think this function would make more sense as a function called `Add` on the type `PVBreakdown`. It would be a shame to want to add two `PVBreakdown` instances and have to put them in `Allocation` instances to do so.